src/app/clusters/localization-configuration-server/localization-configuration-server.cpp

```diff
@@ -123,21 +92,98 @@ CHIP_ERROR LocalizationConfigurationAttrAccess::Read(const ConcreteReadAttribute
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR LocalizationConfigurationAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+bool isSupportedLocales(chip::CharSpan newLangtag, chip::CharSpan & validLangtag)
 {
-    VerifyOrDie(aPath.mClusterId == LocalizationConfiguration::Id);
+    DeviceLayer::AttributeList<chip::CharSpan, DeviceLayer::kMaxLanguageTags> supportedLocales;
 
-    switch (aPath.mAttributeId)
+    if (DeviceLayer::PlatformMgr().GetSupportedLocales(supportedLocales) == CHIP_NO_ERROR)
+    {
+        for (auto locale : supportedLocales)
+        {
+            if (locale.data_equal(newLangtag))
+            {
+                return true;
+            }
+
+            validLangtag = locale;
+        }
+    }
+
+    return false;
+}
+
+} // anonymous namespace
+
+// =============================================================================
+// Pre-change callbacks for cluster attributes
+// =============================================================================
+
+static chip::Protocols::InteractionModel::Status
+emberAfPluginLocalizationConfigurationOnActiveLocaleChange(chip::EndpointId EndpointId, chip::CharSpan newLangtag)
+{
+    chip::Protocols::InteractionModel::Status res;
+    chip::CharSpan validLangtag;
+
+    if (isSupportedLocales(newLangtag, validLangtag))
+    {
+        res = chip::Protocols::InteractionModel::Status::Success;
+    }
+    else
+    {
+        res = chip::Protocols::InteractionModel::Status::InvalidValue;
+    }
+
+    return res;
+}
+
+static chip::Protocols::InteractionModel::Status
+emberAfPluginLocalizationConfigurationOnUnhandledAttributeChange(chip::EndpointId EndpointId, EmberAfAttributeType attrType,
+                                                                 uint16_t attrSize, uint8_t * attrValue)
+{
+    return chip::Protocols::InteractionModel::Status::Success;
+}
+
+chip::Protocols::InteractionModel::Status MatterLocalizationConfigurationClusterServerPreAttributeChangedCallback(
+    const chip::app::ConcreteAttributePath & attributePath, EmberAfAttributeType attributeType, uint16_t size, uint8_t * value)
+{
+    chip::Protocols::InteractionModel::Status res;
+
+    switch (attributePath.mAttributeId)
     {
-    case ActiveLocale::Id:
-        return WriteActiveLocale(aPath.mEndpointId, aDecoder);
+    case chip::app::Clusters::LocalizationConfiguration::Attributes::ActiveLocale::Id: {
+        auto langtag = chip::CharSpan(reinterpret_cast<const char *>(&value[1]), static_cast<size_t>(value[0]));
```

Comment:
```suggestion
        auto langtag = chip::CharSpan(Uint8::from_const_char(&value[1]), static_cast<size_t>(value[0]));
```
and include `CHIPSafeCasts.h`.